recipes/glsl-optimizer/all/test_package/test_package.cpp

```diff
@@ -0,0 +1,8 @@
+#include <glsl_optimizer.h>
+
+int main() {
+    // Simple way to referencing the function to make sure stuff are linked correctly
+    glslopt_ctx*(*ptr)(glslopt_target) = glslopt_initialize;
```

Comment:
Is this actually calling a function?
I'd say this line of code can be an entry for the IOCCC contest.

Reply:
This is creating a function pointer variable named `ptr`, with type `glslopt_ctx(*)(glslopt_target)`, and assign the function `glslopt_initialize` to it. If I'm not mistaken, since this takes address of one of the functions, it will only compile if glsloptimizer is linked correctly.
I don't want to write a full snippet that calls glsloptimizer as it will be very, very long.

Reply:
This is only testing the includes. You are not actually using any symbols.
You really need to call into a function, compiled and linked by the recipe.
This can be very simple.
See https://github.com/aras-p/glsl-optimizer/blob/d78c3d2f249aa870368ad320905bc954c47704f6/src/glsl/glsl_optimizer.h#L66-L67

```c
glslopt_ctx *ctx = glslopt_initialize(kGlslTargetOpenGL);
glslopt_cleanup(ctx);
```

Reply:
Based on my knowledge (taking pointer is ODR-use, which requires symbol definition) and testing (with gcc10 + test.c, test.h, main.c), what I did does check proper linking.
But changed to your suggestion anyways, as it's more straight forward :)

Reply:
Yeah, you fetched the pointer of the initialize function, where the above code calls it.